### PR TITLE
Enable port configurability

### LINKED ISSUE #305: Make API server ports configurable
**Issue or Feature Request:**

Right now, the ports for the API server are hard coded:
https://github.com/bottlerocket-os/bottlerocket-update-operator/blob/1c44ba6e2f4aad17f8b06fd531caf5343d6ccf33/models/src/constants.rs#L59-L60

We should make these configurable so users can have the opportunity to group ports and apply broad network rules to like services

### PR DESCRIPTION
**Pull request by jpmcb**
```
- Api-server: Port can be configured from default 8443
- Agent / webhook: Port can be configured from default 443

Signed-off-by: John McBride <jpmmcb@amazon.com>
```

### Issue number:

Closes: #305

### Description of changes:

Enables ports used by the API server and the agent to be configurable. Removes constant values and instead, interpolates from the container environment.

### Testing done:

Ran through integration testing 👍🏼 

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,43 @@ Be sure to use the git tag for the Update Operator release that you plan to depl
...
+        - command:
+            - "./api-server"
+          env:
+            - name: APISERVER_INTERNAL_PORT
+              value: 999
+      ...
+```
+
+You'll then also need to adjust the various "port" entries in the manifest
+to correctly reflect what port the API server starts on and expects for it's service port:
```
Comment (stmcginnis) [inline README.md]: NIt: s/it's/its/

File: README.md [hunk 2]
```diff
@@ -29,6 +29,43 @@ Be sure to use the git tag for the Update Operator release that you plan to depl
...
+          name: brupop-apiserver
+          namespace: brupop-bottlerocket-aws
+          path: /crdconvert
+          port: 123
+```
+
+The default values are generated from the `.env` file (and can be used when building the update operator from source to configure your own ports):
+- `APISERVER_INTERNAL_PORT = 8443` - This is the container port the API server starts on.
+  If this environment variable is _not_ found, the Brupop API server will fail to start.
+- `APISERVER SERVICE_PORT = 443` - This is the port Brupop's Kubernetes Service uses to target the internal API Server port.
```
Comment (stmcginnis) [inline README.md]: Missing `_` between `APISERVER` and `SERVICE`.

**Review by stmcginnis — COMMENTED** [review]
Just a couple minor things.

File: apiserver/src/client/webclient.rs [hunk 3]
```diff
@@ -65,13 +65,23 @@ pub trait APIServerClient {
...
 impl K8SAPIServerClient {
-    pub fn new(k8s_projected_token_path: String) -> Self {
-        Self {
+    pub fn new(k8s_projected_token_path: String) -> Result<Self> {
+        let service_port: i32 = env::var(APISERVER_SERVICE_PORT_ENV_VAR)
+            .context(error::MissingEnvVariableSnafu {
+                variable: APISERVER_SERVICE_PORT_ENV_VAR.to_string(),
+            })?
+            .parse()
+            .unwrap();
```
Comment (gthao313) [inline apiserver/src/client/webclient.rs]: I think better to avoid using `unwrap`

File: apiserver/src/main.rs [hunk 4]
```diff
@@ -37,9 +37,17 @@ async fn run_server() -> Result<(), apiserver_error::Error> {
         .await
         .context(apiserver_error::K8sClientCreateSnafu)?;
 
+    let internal_port: i32 = env::var(APISERVER_INTERNAL_PORT_ENV_VAR)
+        .context(apiserver_error::MissingEnvVariableSnafu {
+            variable: APISERVER_INTERNAL_PORT_ENV_VAR.to_string(),
+        })?
+        .parse()
+        .unwrap();
```
Comment (gthao313) [inline apiserver/src/main.rs]: Same as above

**Comment by gthao313** [comment]
Test looks good on my side.

**Comment by jpmcb** [comment]
Force pushed to address above nits/comments! ✍🏼

**Review by gthao313 — APPROVED** [review]

**Review by stmcginnis — APPROVED** [review]
Looks good!

**Pull request merged**